include/alpaka/mem/fence/MemFenceCpu.hpp

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Concepts.hpp>
+#include <alpaka/mem/fence/Traits.hpp>
+
+#include <atomic>
+
+namespace alpaka
+{
+    //! The default CPU memory fence.
+    class MemFenceCpu : public concepts::Implements<ConceptMemFence, MemFenceCpu>
+    {
+    public:
+        MemFenceCpu() = default;
+        MemFenceCpu(MemFenceCpu const&) = delete;
+        auto operator=(MemFenceCpu const&) -> MemFenceCpu& = delete;
+        MemFenceCpu(MemFenceCpu&&) = delete;
+        auto operator=(MemFenceCpu&&) -> MemFenceCpu& = delete;
+        ~MemFenceCpu() = default;
+    };
```

Comment:
Why do we need to delete all the copy/move operations? This is just a tag type we mixin into the accelerator.
```suggestion
    struct MemFenceCpu : public concepts::Implements<ConceptMemFence, MemFenceCpu> {};
```

Reply:
I just mirrored what we are doing elsewhere in alpaka (see the intrinsics for example). But perhaps we should review the other instances of this, too.

Reply:
FYI for intrinsics I also mirrored how it was done elsewhere, without much thinking on that

Reply:
Thanks for the info. Guess we'll have to do an alpaka code review at some point. However, this is out of scope for this PR.

Reply:
Then please fork off an issue next time! There is a button right here in the `...` menu on github.
I did it now: #1396